docs/rfc/007-integer64-type.md

```diff
@@ -0,0 +1,75 @@
+# RFC: Conjure 64-bit Integer Type
+
+2019-02-14
+
+_The 54-bit Integer is Not Enough_
+
+Conjure currently supports three numeric types: `integer`, `double`, and `safelong`. Unfortunately for many use-cases, 2<sup>53</sup> - 1 is not large enough.
+
+These cases are relatively common and often result in one of three failure scenarios:
+1. Use safelong because it exists. We often measure nanoseconds, however a safelong can only represent three months of nanoseconds, which is long enough that it's unlikely to be caught in testing.
+1. Represent 64-bit integer values as strings in conjure. This is only as correct as the application code which sets and parses values, which is precisely the problem Conjure is built to solve. In this case the conjure definition fails to accurately describe the wire API.
+1. Use external type imports to polyfill the 64-bit integer type. This is incredibly dangerous, often resulting in code that behaves inconsistently between languages.
+
+## Proposal
+
+New conjure `integer64` type semantics matching `integer` except for the allowed values, and JSON serialized form.
```

Comment:
Bikeshed: should we use `long` as the name? We match Java's naming for integer, boolean, double currently.

Reply:
I went back and forth between the two, I don't have a strong preference. I think `integer<bits>` is a bit cleaner if we ever need to add larger integer types, and allows us to alias `integer` to `integer32` and `safelong` to `integer54`.

Reply:
I'd be on board with moving to `integer32`, `float64`, etc.